slkspec/core.py

```diff
@@ -151,6 +151,18 @@ def _retrieve_items(self, retrieve_files: list[tuple[str, str]]) -> None:
             for out_file in map(Path, inp_files):
                 (output_dir / out_file.name).chmod(self.file_permissions)
 
+    def mkdirs(self, path):
+        rp = os.path.realpath(path)
+        if os.access(rp, os.F_OK):
```

Comment:
From the [python documentation](https://docs.python.org/3/library/os.html#os.access) it seems that `os.access` is not a very good function to use. Can we use an alternative?

Reply:
My feeling is that these problems concern problems that we don't really need to care about.
This is really just for checking if the directory is there, or we need to create it.

Reply:
We might want to explicitly check for write access though, and throw an exception if the directory exists, but we can't write.